src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -2046,7 +2046,7 @@ Method* SystemDictionary::find_method_handle_intrinsic(vmIntrinsicID iid,
 
   assert(spe != NULL && spe->method() != NULL, "");
   assert(Arguments::is_interpreter_only() || (spe->method()->has_compiled_code() &&
-         spe->method()->code()->entry_point() == spe->method()->from_compiled_entry()),
+         spe->method()->code()->code_begin() == spe->method()->from_compiled_entry()),
```

Comment:
Not sure if this assert is still needed/useful, since we should only encounter MH linker blobs here, for which the `from_compiled_entry` is always `code_begin`.

I'd suggest just removing this assert.